library/general/src/lib/yast2/rel_url.rb

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "uri"
+
+Yast.import "InstURL"
+
+module Yast2
+  # Class for working with relative URLs ("relurl://")
+  class RelURL
+    attr_reader :base, :relative
+
+    # Is the URL a relative URL?
+    #
+    # @param url [String, URI] the URL
+    # @return [Boolean] `true` if the URL uses the "relurl" schema, otherwise `false`
+    def self.relurl?(url)
+      URI(url).scheme == "relurl"
+    end
+
+    # Create RelURL object with URL relative to the installation repository
+    #
+    # @param rel_url [String, URI] the relative URL
+    # @return [RelURL]
+    #
+    # @note Works properly only during installation/upgrade, do not use
+    #   in an installed system.
+    def self.from_installation_repository(rel_url)
+      base_url = Yast::InstURL.installInf2Url("")
+      new(base_url, rel_url)
```

Comment:
what will happen if nont relative url is called?

Reply:
The result will be this non-relative URL. The caller is responsible to use a relative URL to make it work properly.